src/main/java/org/sciborgs1155/robot/Hopper/Hopper.java

```diff
@@ -0,0 +1,55 @@
+package org.sciborgs1155.robot.Hopper;
+
+import static org.sciborgs1155.robot.Hopper.HopperConstants.*;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import monologue.Logged;
+import monologue.Monologue.LogBoth;
+import org.sciborgs1155.robot.Robot;
+
+public class Hopper extends SubsystemBase implements Logged {
+
+  final PIDController pid = new PIDController(kp, ki, kd);
```

Comment:
i take back whatever i said that caused you to pass in the pid to the hopper -- my bad. The control should be done here. 

If you were to do the pid stuff in the hopper, which would also work, instead of passing in the controller, you would pass in the constants. It's bad practice to have the pid (or anything mutable) accessible in both classes.

Instead, for this just have the pid here, and update the voltage in periodic using the controller